### PR TITLE
Assist in the bootstrapping process of the rusk-driven devnet

### LINKED ISSUE #990: Assist in the bootstrapping process of the rusk-driven devnet
**Describe the bug**
Introduce changes that makes it easier  to bootstrap and monitor a rusk network on DO. More particularly, this issue should touch configs, improve logging,  extend error handling etc.

### LINKED ISSUE #1002: consensus: Reset reduction phase state on initialization
**Describe the bug**
If Reduction step is not initialized with a new candidate block, it reuses the last one provided from previous iteration. As a result of that, a provisioner may vote for a wrong candidate hash, rather than the present one.

**Expected behaviour**
Any internal state kept by Reduction phase other than `timeout` setting should be reset before running the step.

### LINKED ISSUE #1003: consensus: Filter out agreement messages with erroneous step votes
**Describe the bug**
Consensus could spread an agreement message containing erroneous step votes (such as lacking a signature, hash or having a bitset=0 ). This situation arises when the step votes from the first reduction are flawed, while those from the second reduction are valid.

**Solution**
Follow the example of `stepVotesAreValid` usage in the golang impl.

### PR DESCRIPTION
**Pull request by goshawk-3**
fixes #990 
fixes #1002
fixes #1003

Major improvements with this PR:

- Reset reduction state on step initialization
- Filter out agreement messages with invalid first/second reduction 
- Selection phase does not double its timeout settings
- Make default CONSENSUS_TIMEOUT_MS=20s
- Extend logging details in consensus

### CODE REVIEW

**Comment by herr-seppia** [comment]
@goshawk-3 
Since I had to properly split commits, I added you as coauthor of the new comiits

File: consensus/src/agreement/accumulator.rs [hunk 1]
```diff
@@ -207,11 +201,11 @@ impl Accumulator {
             if *agr_weight >= target_quorum {
                 info!(
                     event = "quorum reached",
-                    hash = hdr.block_hash.encode_hex::<String>(),
+                    hash = to_str(&hdr.block_hash),
                     msg_round = hdr.round,
                     msg_step = hdr.step,
                     target = target_quorum,
-                    aggr_count = agr_weight
+                    agr_weight = agr_weight
```
Comment (herr-seppia) [inline consensus/src/agreement/accumulator.rs:208]: nit: you can remove the `= agr_weight`

File: consensus/src/agreement/verifiers.rs [hunk 2]
```diff
@@ -110,9 +151,18 @@ pub async fn verify_votes(
...
+        let total = guard.total_occurrences(&sub_committee, cfg);
+
+        if total < target_quorum {
+            tracing::error!(
+                desc = "vote_set_too_small",
+                committee = format!("{:#?}", sub_committee),
+                cfg = format!("{:#?}", cfg),
+                bitset = bitset,
+                target_quorum = target_quorum,
+                total = total,
```
Comment (herr-seppia) [inline consensus/src/agreement/verifiers.rs:163]: Ditto

File: consensus/src/phase.rs [hunk 3]
```diff
@@ -47,21 +47,22 @@ pub enum Phase<T: Operations, D: Database> {
 
 impl<T: Operations + 'static, D: Database + 'static> Phase<T, D> {
     pub fn initialize(&mut self, msg: &Message, round: u64, step: u8) {
-        trace!(
-            "init phase:{} with msg {:?} at round:{} step:{}",
+        debug!(
+            "event: init_step:{} round:{} step:{} msg: {:#?}",
```
Comment (herr-seppia) [inline consensus/src/phase.rs:51]: Should you change this to be formatted the same as the rest?
I mean with `event = "init_step", round, step, msg = format!("{msg:#?}")`

File: rusk-recovery/config/testnet.toml [hunk 4]
```diff
@@ -134,3 +135,113 @@ amount = 500_000_000_000_000
...
+amount = 500_000_000_000_000
+ 
+
+
+
+
+
+
+
+
```
Comment (herr-seppia) [inline rusk-recovery/config/testnet.toml:247]: nit: remove these lines

File: node/Cargo.toml [hunk 5]
```diff
@@ -32,6 +32,7 @@ blake2 = "0.10.5"
 dusk-bls12_381-sign = { version = "0.4", default-features = false }
 console-subscriber = { version = "0.1.8", optional = true }
 smallvec = "1.10.0"
+bs58 = "0.4"
```
Comment (herr-seppia) [inline node/Cargo.toml:35]: I believe this is not required

**Review by herr-seppia — APPROVED** [review]
LGTM

just some nits

File: consensus/src/phase.rs (hunk 3 above)
Comment (goshawk-3) [reply consensus/src/phase.rs:51]: Indeed, this is covered in next issue.

File: consensus/src/agreement/verifiers.rs (hunk 2 above)
Comment (goshawk-3) [reply consensus/src/agreement/verifiers.rs:163]: Indeed, this will be also covered with next issue.

**Pull request merged**